File: trunk/document/bar code reader/barcode/Examples/WindowsFoms_CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

using Lesnikowski.Barcode;

namespace WindowsForms_CS
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            BaseBarcode b = BarcodeFactory.GetBarcode(Symbology.I2of5);
            b.Number = "123456789";
            b.ChecksumAdd = true;
            b.Rotation = RotationType.Degrees90;
            pictureBox1.Image = b.Render();
        }

        #region Print

        private void printDocument1_PrintPage(object sender, System.Drawing.Printing.PrintPageEventArgs e)
        {
            this.barcodeControl1.Render(e.Graphics, 100, 100);

            BaseBarcode b = BarcodeFactory.GetBarcode(Symbology.I2of5);
            b.Number = "123456789";
            b.ChecksumAdd = true;
            b.Rotation = RotationType.Degrees90;

            // 92 dots per inch (screen resolution) * 1.5 cm /2.54 inch
            b.Height = (int)(92.0F * 1.5 / 2.54);            // 1.5 cm
            b.NarrowBarWidth = (int)(92.0F * 0.05 / 2.54);   // 0.5 mm
            b.Render(e.Graphics, 300, 100);
        }

        private void printToolStripMenuItem_Click(object sender, EventArgs e)
        {
            if (this.printDialog1.ShowDialog() == DialogResult.OK)
            {
                this.printDialog1.Document.Print();
            }
        }

        private void printPreviewToolStripMenuItem_Click(object sender, EventArgs e)
        {
            this.printPreviewDialog1.ShowDialog();
        }

        #endregion

    };
}

File: trunk/document/bar code reader/barcode/Examples/WindowsFoms_VB/Form1.vb
Imports Lesnikowski.Barcode

Public Class Form1

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        Dim b As BaseBarcode
        b = BarcodeFactory.GetBarcode(Symbology.I2of5)

        b.Number = "123456789"
        b.ChecksumAdd = True
        b.Rotation = RotationType.Degrees90
        pictureBox1.Image = b.Render()
    End Sub

    Private Sub printToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles printToolStripMenuItem.Click
        If printDialog1.ShowDialog() = Windows.Forms.DialogResult.OK Then
            printDialog1.Document.Print()
        End If
    End Sub

    Private Sub printPreviewToolStripMenuItem_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles printPreviewToolStripMenuItem.Click
        printPreviewDialog1.ShowDialog()
    End Sub

    Private Sub printDocument1_PrintPage(ByVal sender As System.Object, ByVal e As System.Drawing.Printing.PrintPageEventArgs) Handles printDocument1.PrintPage
        barcodeControl1.Render(e.Graphics, 100, 100)

        Dim b As BaseBarcode
        b = BarcodeFactory.GetBarcode(Symbology.I2of5)

        b.Number = "123456789"
        b.ChecksumAdd = True
        b.Rotation = RotationType.Degrees90
        '92 dots per inch (screen resolution) * 1.5 cm /2.54 inch
        b.Height = 92.0F * 1.5 / 2.54            '1.5 cm
        b.NarrowBarWidth = 92.0F * 0.05 / 2.54   '0.5 mm
        b.Render(e.Graphics, 300, 100)
    End Sub
End Class
